share/doc/wake/logging.md

```diff
@@ -0,0 +1,152 @@
+# Logging & Console Output
+
+This article gives a high level overview of how wake logging works,
+and other things that impact what you see on the console when wake is executing or running database commands.
+Check out [print.wake](https://github.com/sifive/wake/blob/master/share/wake/lib/core/print.wake)
+for commented code.
+
+## Loggers & Log Levels
+
+Wake has a concept of loggers,
+to which text can be written during wake execution by various sources.
+A logger combines a string log level (e.g. "error", "info", "my_custom_level")
+and a display format (color and intensity).
+There can be multiple loggers with the same log level, though this is
+not a common use case.
+
+|Term Used In this Document | What wake code/libraries call it |
+----------------------------|----------------------------------|
+|  logger                   | `LogLevel`                       |
+|  log level                | `LogLevel.name`                  |
+
+## Using & Creating Loggers
+
+`wake` has a number of built in loggers in which all start with `log...`.
+You can also make your own logger using `mkLogLevel`.
+You can use this to customize the color you want it to be shown as (if at all)
+and give more fine-grained control over which wake output stream it should go to.
+
+
+| wake `def`        | Color/Intensity | log level |
+|-------------------|-----------------|-----------|
+| logDebug          |     Blue        | "debug"   |
+| logInfo           |     Dim         | "info"    |
+| logEcho           |      N/A        | "echo"    |
+| logReport         |    Magenta      | "report"  |
+| logWarning        |    Yellow       | "warning" |
+| logError          |     Red         | "error"   |
+| logNever          |     N/A         | "null"    |
+| mkLogLevel "foo" (Some Green) |  Green     | "foo"     |
+| mkLogLevel2 "foo" (Some Green) (Some Bright) | Bright Green   | "foo"      |
```

Comment:
```suggestion
| mkLogLevel2 "bar" (Some Green) (Some Bright) | Bright Green   | "bar"      |
```

if the colors aren't well supported with multiple loggers with the same "log level" aka name string using different colors, let's not put it here

Reply:
Oh, I didn't even consider that having the same name could be read as defining these in the same codebase; I just figured it was "insert your name here subject to the other described restrictions".  Good fix!